### PR TITLE
Validate cross-referenced parameters between PostgresCluster and PostgresClusterClass

### PR DESCRIPTION
**Pull request by limak9182**
### Description

  - Adds admission-time cross-resource validation for PostgresCluster against its referenced PostgresClusterClass
  - Uses `mgr.GetAPIReader()` (live API read, not cached) to avoid false rejections during cache propagation

### Key Changes

  | Rule | Field | Condition |
  |------|-------|-----------|
  | Class existence | `spec.class` | Referenced PostgresClusterClass must exist |
  | Storage floor | `spec.storage` | Cannot be lower than class default |
  | Version floor | `spec.postgresVersion` | Major version cannot be lower than class default. If minor version specified in class cluster's minor cannot be lower. |
  | Pooler enablement | `spec.connectionPoolerEnabled` | Cannot be `true` when class has it disabled |
  | Merge completeness | `spec.instances`, `spec.postgresVersion`, `spec.storage` | After overlay, all three must be resolvable from cluster or class |

### Testing and Verification

  - unit test cases with fake `client.Reader` covering all rules and edge cases
  - Existing pg_hba validation tests updated to pass `nil` reader (no behavioral change)
  - `go build ./...` passes
  - `go test ./pkg/splunk/enterprise/validation/...` passes

### Related Issues

_Jira tickets, GitHub issues, Support tickets..._

### PR Checklist

- [ ] Code changes adhere to the project's coding standards.
- [ ] Relevant unit and integration tests are included.
- [ ] Documentation has been updated accordingly.
- [ ] All tests pass locally.
- [ ] The PR description follows the project's guidelines.

### CODE REVIEW

File: pkg/postgresql/cluster/adapter/webhook/postgres_webhook_integration_test.go [hunk 1]
```diff
@@ -557,3 +559,166 @@ func TestPostgresClusterClassPgHBAUpdateIntegration(t *testing.T) {
...
+			obj: &enterpriseApi.PostgresCluster{
+				TypeMeta:   metav1.TypeMeta{APIVersion: "enterprise.splunk.com/v4", Kind: "PostgresCluster"},
+				ObjectMeta: metav1.ObjectMeta{Name: "test", Namespace: "default"},
+				Spec:       enterpriseApi.PostgresClusterSpec{Class: "nonexistent"},
+			},
+			wantAllowed: false,
+			wantMessage: "PostgresClusterClass not found",
+		},
+		{
+			name: "rejected - storage below class floor",
```
Comment (DmytroPI-dev) [inline pkg/postgresql/cluster/adapter/webhook/postgres_webhook_integration_test.go]: `storage below class floor` - I assume we can override this value on a cluster lever even with lower ones, am I wrong?

File: pkg/postgresql/cluster/adapter/webhook/postgres_webhook_integration_test.go [hunk 2]
```diff
@@ -557,3 +559,166 @@ func TestPostgresClusterClassPgHBAUpdateIntegration(t *testing.T) {
...
+				Spec: enterpriseApi.PostgresClusterSpec{
+					Class:   "prod",
+					Storage: ptrQuantity("10Gi"),
+				},
+			},
+			wantAllowed: false,
+			wantMessage: "storage cannot be lower than class default",
+		},
+		{
+			name: "rejected - version below class floor",
```
Comment (DmytroPI-dev) [inline pkg/postgresql/cluster/adapter/webhook/postgres_webhook_integration_test.go:636]: `version below class floor` - same as above - can't we set lower versions on cluster level now?

File: pkg/postgresql/cluster/adapter/webhook/postgres_webhook_integration_test.go [hunk 3]
```diff
@@ -557,3 +559,166 @@ func TestPostgresClusterClassPgHBAUpdateIntegration(t *testing.T) {
...
+			obj: &enterpriseApi.PostgresCluster{
+				TypeMeta:   metav1.TypeMeta{APIVersion: "enterprise.splunk.com/v4", Kind: "PostgresCluster"},
+				ObjectMeta: metav1.ObjectMeta{Name: "test", Namespace: "default"},
+				Spec: enterpriseApi.PostgresClusterSpec{
+					Class:                   "prod",
+					ConnectionPoolerEnabled: ptrBool(true),
+				},
+			},
+			wantAllowed: false,
+			wantMessage: "connectionPoolerEnabled cannot be enabled",
```
Comment (DmytroPI-dev) [inline pkg/postgresql/cluster/adapter/webhook/postgres_webhook_integration_test.go]: "connectionPoolerEnabled cannot be enabled" - sounds strange (enabled cannot be enabled 😸 )

File: pkg/postgresql/cluster/adapter/webhook/postgres_webhook_integration_test.go [hunk 4]
```diff
@@ -557,3 +559,166 @@ func TestPostgresClusterClassPgHBAUpdateIntegration(t *testing.T) {
...
+		},
+		{
+			name: "allowed - storage equal to class",
```
Comment (DmytroPI-dev) [inline pkg/postgresql/cluster/adapter/webhook/postgres_webhook_integration_test.go]: "allowed - storage equal to class" - and if higher?

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go [hunk 5]
```diff
@@ -36,12 +40,111 @@ func ValidatePostgresClusterCreate(obj *enterpriseApi.PostgresCluster) field.Err
...
+		allErrs = append(allErrs, validateAgainstClass(obj, reader)...)
+	}
+
 	return allErrs
 }
 
 // ValidatePostgresClusterUpdate validates a PostgresCluster on UPDATE.
-func ValidatePostgresClusterUpdate(obj, oldObj *enterpriseApi.PostgresCluster) field.ErrorList {
-	return ValidatePostgresClusterCreate(obj)
+func ValidatePostgresClusterUpdate(obj, oldObj *enterpriseApi.PostgresCluster, reader client.Reader) field.ErrorList {
```
Comment (DmytroPI-dev) [inline pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go:54]: `oldObj` is ignored on purpose?

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go [hunk 6]
```diff
@@ -36,12 +40,111 @@ func ValidatePostgresClusterCreate(obj *enterpriseApi.PostgresCluster) field.Err
...
+		allErrs = append(allErrs, field.Invalid(
+			field.NewPath("spec").Child("class"),
+			obj.Spec.Class,
+			"referenced PostgresClusterClass not found"))
+		return allErrs
+	}
+
+	classConfig := class.Spec.Config
+
+	mergedInstances := obj.Spec.Instances
```
Comment (DmytroPI-dev) [inline pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go]: do we need separate vars for each value? why not struct?

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go [hunk 7]
```diff
@@ -36,12 +40,111 @@ func ValidatePostgresClusterCreate(obj *enterpriseApi.PostgresCluster) field.Err
...
+	if classConfig == nil {
+		return allErrs
+	}
+
+	if obj.Spec.Storage != nil && classConfig.Storage != nil {
+		if obj.Spec.Storage.Cmp(*classConfig.Storage) < 0 {
+			allErrs = append(allErrs, field.Invalid(
+				field.NewPath("spec").Child("storage"),
+				obj.Spec.Storage.String(),
+				"storage cannot be lower than class default ("+classConfig.Storage.String()+")"))
```
Comment (DmytroPI-dev) [inline pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go]: As above: why we can't override storage on cluster level? I assume it is better to check whether it was changed/removed from spec later on and/or set to lower values. WDYT?

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go [hunk 8]
```diff
@@ -36,12 +40,111 @@ func ValidatePostgresClusterCreate(obj *enterpriseApi.PostgresCluster) field.Err
...
+		clusterMajor, clusterMinor := parseVersion(*obj.Spec.PostgresVersion)
+		classMajor, classMinor := parseVersion(*classConfig.PostgresVersion)
+		if clusterMajor > 0 && classMajor > 0 {
+			versionTooLow := clusterMajor < classMajor ||
+				(clusterMajor == classMajor && classMinor >= 0 && clusterMinor >= 0 && clusterMinor < classMinor)
+			if versionTooLow {
+				allErrs = append(allErrs, field.Invalid(
+					field.NewPath("spec").Child("postgresVersion"),
+					*obj.Spec.PostgresVersion,
+					"postgresVersion cannot be lower than class default ("+*classConfig.PostgresVersion+")"))
```
Comment (DmytroPI-dev) [inline pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go]: Same as for storage size, why can't we set lower version on cluster level? And not checking removal/downgrading later on?

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go [hunk 9]
```diff
@@ -36,12 +40,111 @@ func ValidatePostgresClusterCreate(obj *enterpriseApi.PostgresCluster) field.Err
...
+func parseVersion(version string) (major, minor int) {
+	for i, ch := range version {
+		if ch == '.' {
+			major, _ = strconv.Atoi(version[:i])
+			minor, _ = strconv.Atoi(version[i+1:])
+			return major, minor
+		}
+	}
+	major, _ = strconv.Atoi(version)
+	return major, -1
```
Comment (DmytroPI-dev) [inline pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go:132]: Do we need to check for minor at all? I mean, we are allowed to downgrade minor version, don't we?

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation_test.go [hunk 10]
```diff
@@ -190,3 +196,344 @@ func TestGetPostgresClusterWarningsOnUpdate(t *testing.T) {
...
+	for _, obj := range objects {
+		b = b.WithRuntimeObjects(obj)
+	}
+	return b
+}
+
+func ptrBool(b bool) *bool       { return &b }
+func ptrString(s string) *string { return &s }
+func ptrInt32(i int32) *int32    { return &i }
+
```
Comment (DmytroPI-dev) [inline pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation_test.go]: why don't use `	"k8s.io/utils/ptr"` instead of custom methods?

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation_test.go [hunk 11]
```diff
@@ -190,3 +196,344 @@ func TestGetPostgresClusterWarningsOnUpdate(t *testing.T) {
...
+			obj: &enterpriseApi.PostgresCluster{
+				Spec: enterpriseApi.PostgresClusterSpec{
+					Class:           "prod-pinned",
+					PostgresVersion: ptrString("17.5"),
+				},
+			},
+			wantErrCount: 0,
+		},
+		{
+			name:  "invalid - cluster minor lower than class minor",
```
Comment (DmytroPI-dev) [inline pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation_test.go:359]: We are allowed to downgrade minors, aren't we?

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go [hunk 12]
```diff
@@ -36,12 +40,111 @@ func ValidatePostgresClusterCreate(obj *enterpriseApi.PostgresCluster) field.Err
...
+			field.NewPath("spec").Child("class"),
+			obj.Spec.Class,
+			"referenced PostgresClusterClass not found"))
+		return allErrs
+	}
+
+	classConfig := class.Spec.Config
+
+	mergedInstances := obj.Spec.Instances
+	mergedVersion := obj.Spec.PostgresVersion
```
Comment (DmytroPI-dev) [inline pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go]: Also, we are re-inventing the wheel here 😸 , as we have class/cluster merge logic (instances, storage, postgresVersion, pooler defaults, et.) separately from the reconcile path. That’s easy to drift over time. I’d extract a shared helper if possible so admission and reconciliation enforce the same rules from one place.

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go [hunk 13]
```diff
@@ -36,12 +40,111 @@ func ValidatePostgresClusterCreate(obj *enterpriseApi.PostgresCluster) field.Err
...
+		if classDisabled {
+			allErrs = append(allErrs, field.Invalid(
+				field.NewPath("spec").Child("connectionPoolerEnabled"),
+				true,
+				"connectionPoolerEnabled cannot be enabled when disabled in PostgresClusterClass"))
+		}
+	}
+
+	return allErrs
+}
```
Comment (DmytroPI-dev) [inline pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go]: This changes `connectionPoolerEnabled` from an override into a hard class policy. The API comments currently say the cluster value “takes precedence” / “can be overridden”, so rejecting `cluster=true` when the class is `false` looks like a semantic change that should either be documented explicitly or revisited.

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go [hunk 14]
```diff
@@ -36,12 +40,111 @@ func ValidatePostgresClusterCreate(obj *enterpriseApi.PostgresCluster) field.Err
...
+	if mergedStorage == nil {
+		allErrs = append(allErrs, field.Required(specPath.Child("storage"),
+			"must be set in PostgresCluster or PostgresClusterClass"))
+	}
+
+	if classConfig == nil {
+		return allErrs
+	}
+
+	if obj.Spec.Storage != nil && classConfig.Storage != nil {
```
Comment (DmytroPI-dev) [inline pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go]: Same point for `storage` and `postgresVersion`: the merge code treats class values as defaults, but this webhook now treats them as minimum allowed values. If that policy shift is intentional, I’d update the API docs/comments because “overrides” reads differently today.

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go [hunk 15]
```diff
@@ -36,12 +40,111 @@ func ValidatePostgresClusterCreate(obj *enterpriseApi.PostgresCluster) field.Err
...
+	var allErrs field.ErrorList
+
+	class := &enterpriseApi.PostgresClusterClass{}
+	if err := reader.Get(context.Background(), client.ObjectKey{Name: obj.Spec.Class}, class); err != nil {
+		allErrs = append(allErrs, field.Invalid(
+			field.NewPath("spec").Child("class"),
+			obj.Spec.Class,
+			"referenced PostgresClusterClass not found"))
+		return allErrs
+	}
```
Comment (DmytroPI-dev) [inline pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go:72]: `reader.Get()` collapses every error into “referenced PostgresClusterClass not found”. That will hide real issues like API/RBAC failures and return a misleading validation error. I’d special-case IsNotFound and surface other errors separately.

File: pkg/splunk/enterprise/validation/validate.go [hunk 16]
```diff
@@ -84,10 +85,10 @@ func ValidateWithClient(ar *admissionv1.AdmissionReview, validators map[schema.G
...
 	}
 
-	// Create validation context if client is available
+	// Create validation context if client or reader is available
 	var vc *ValidationContext
-	if k8sClient != nil {
-		vc = NewValidationContext(k8sClient, req.Namespace)
+	if k8sClient != nil || reader != nil {
+		vc = NewValidationContext(k8sClient, reader, req.Namespace)
 	}
```
Comment (DmytroPI-dev) [inline pkg/splunk/enterprise/validation/validate.go]: Cross-resource validation now depends entirely on `vc.Reader.` If a caller provides only a cached client, or calls the validator directly, the checks silently disappear. I’d prefer a Reader -> Client fallback so the behavior is less wiring-sensitive.

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation_test.go [hunk 17]
```diff
@@ -190,3 +196,344 @@ func TestGetPostgresClusterWarningsOnUpdate(t *testing.T) {
...
+					Instances:       ptrInt32(1),
+					PostgresVersion: ptrString("17"),
+					Storage:         ptrQuantity("10Gi"),
+				},
+			},
+			wantErrCount: 0,
+		},
+	}
+
+	for _, tt := range tests {
```
Comment (DmytroPI-dev) [inline pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation_test.go:514]: The new tests cover the happy path well, but they mostly assert error count/field or Allowed. I’d add message assertions for the unit tests and one update-path integration case for the cross-resource checks, so semantic regressions in the comparison logic are easier to catch.

File: pkg/postgresql/cluster/adapter/webhook/postgres_webhook_integration_test.go [hunk 18]
```diff
@@ -557,3 +559,166 @@ func TestPostgresClusterClassPgHBAUpdateIntegration(t *testing.T) {
...
+	s := runtime.NewScheme()
+	enterpriseApi.AddToScheme(s)
+	b := fake.NewClientBuilder().WithScheme(s)
+	for _, obj := range objects {
+		b = b.WithRuntimeObjects(obj)
+	}
+	return b
+}
+
+func TestCrossResourceValidationIntegration(t *testing.T) {
```
Comment (DmytroPI-dev) [inline pkg/postgresql/cluster/adapter/webhook/postgres_webhook_integration_test.go:588]: The new tests cover the happy path well, but they mostly assert error count/field or Allowed. I’d add message assertions for the unit tests and one update-path integration case for the cross-resource checks, so semantic regressions in the comparison logic are easier to catch.

File: cmd/main.go [hunk 19]
```diff
@@ -348,6 +348,7 @@ func main() {
 			ReadTimeout:  readTimeout,
 			WriteTimeout: writeTimeout,
 			Client:       mgr.GetClient(),
+			Reader:       mgr.GetAPIReader(),
```
Comment (mploski) [inline cmd/main.go]: could you elab more, why we need this reader? What problem do we solve?

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go [hunk 20]
```diff
@@ -36,12 +40,111 @@ func ValidatePostgresClusterCreate(obj *enterpriseApi.PostgresCluster) field.Err
 		}
 	}
 
+	if reader != nil {
```
Comment (mploski) [inline pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go:46]: could you elab why we expect some reader to validate against class logic?

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go (hunk 12 above)
Comment (mploski) [reply pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go]: +1

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go (hunk 14 above)
Comment (mploski) [reply pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go]: I think there is no contradiction here i.e our default logic is there and it is used in the runtime reconciler but business rules validation is done here in the validation webhook. So we have "overrides" with guardrails in place. I agree we need to document this logic properly though

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go (hunk 7 above)
Comment (mploski) [reply pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go]: I agree with @DmytroPI-dev our idea here is not to say: if class has 1000GB's of disc our cluster cannot use 999. What we really want to say once we DEPLOYED cluster with 1000 we cannot change cluster config to 999

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go (hunk 8 above)
Comment (mploski) [reply pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go]: In this case we probably should enforce minimum version of postgres coming from class and not allow lowering it down from the compliance perspective

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go (hunk 13 above)
Comment (mploski) [reply pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go]: +1 from connection pooler perspective I think we SHOULD allow override it from false to positive BUT if user make it enabled in cluster we need to enforce that cnpg config for pooler is in place otherwise reject

File: pkg/splunk/enterprise/validation/context.go [hunk 21]
```diff
@@ -27,9 +27,14 @@ import (
...
 // to access Kubernetes resources
 type ValidationContext struct {
-	// Client is the Kubernetes client for resource lookups
+	// Client is the Kubernetes client for resource lookups (cached)
 	Client client.Client
 
+	// Reader is a live API reader that bypasses the cache.
+	// Use this for cross-resource lookups where eventual consistency
+	// could cause false rejections.
+	Reader client.Reader
```
Comment (mploski) [inline pkg/splunk/enterprise/validation/context.go]: Is this really a problem in our case? if we for example deploy new cluster CR and we need to check latest deployed clusterClass isnt it already populated within the cache? Why we need to bypass it? Why this isnt a problem for other CR's checked by this?

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go (hunk 7 above)
Comment (limak9182) [reply pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go]: make sense, in that case we can rely on our already implemented CEL validation:
```go
  // +kubebuilder:validation:XValidation:rule="!has(oldSelf.storage) || (has(self.storage) && quantity(self.storage).compareTo(quantity(oldSelf.storage)) >= 0)",messageExpression="..."
```

I removed this block. Now, I'm checking only if we have storage defined at least onece in Class or Cluster

File: pkg/postgresql/cluster/adapter/webhook/postgres_webhook_integration_test.go (hunk 1 above)
Comment (limak9182) [reply pkg/postgresql/cluster/adapter/webhook/postgres_webhook_integration_test.go]: yes, correct

File: pkg/postgresql/cluster/adapter/webhook/postgres_webhook_integration_test.go (hunk 2 above)
Comment (limak9182) [reply pkg/postgresql/cluster/adapter/webhook/postgres_webhook_integration_test.go:636]: For the compliance we prohibit setting lower version than in class

File: pkg/postgresql/cluster/adapter/webhook/postgres_webhook_integration_test.go (hunk 3 above)
Comment (limak9182) [reply pkg/postgresql/cluster/adapter/webhook/postgres_webhook_integration_test.go]: fixed

File: pkg/postgresql/cluster/adapter/webhook/postgres_webhook_integration_test.go (hunk 4 above)
Comment (limak9182) [reply pkg/postgresql/cluster/adapter/webhook/postgres_webhook_integration_test.go]: removed

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go (hunk 5 above)
Comment (limak9182) [reply pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go:54]: yes, we don't care about oldObj

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go (hunk 6 above)
Comment (limak9182) [reply pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go]: refactored

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go (hunk 8 above)
Comment (limak9182) [reply pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go]: done

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go (hunk 9 above)
Comment (limak9182) [reply pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go:132]: resolved, https://github.com/splunk/splunk-operator/pull/1858#discussion_r3130597351, downgrading while being alligned with version set in class is allowed.

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation_test.go (hunk 10 above)
Comment (limak9182) [reply pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation_test.go]: fixed

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation_test.go (hunk 11 above)
Comment (limak9182) [reply pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation_test.go:359]: downgrade is allowed, only setting lower than in class is banned

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go (hunk 12 above)
Comment (limak9182) [reply pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go]: refactored

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go (hunk 13 above)
Comment (limak9182) [reply pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go]: refactored

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go (hunk 14 above)
Comment (limak9182) [reply pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go]: added comment for clarity

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go (hunk 15 above)
Comment (limak9182) [reply pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go:72]: removed API reader

File: pkg/splunk/enterprise/validation/validate.go (hunk 16 above)
Comment (limak9182) [reply pkg/splunk/enterprise/validation/validate.go]: removed API reader

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation_test.go (hunk 17 above)
Comment (limak9182) [reply pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation_test.go:514]: done

File: pkg/postgresql/cluster/adapter/webhook/postgres_webhook_integration_test.go (hunk 18 above)
Comment (limak9182) [reply pkg/postgresql/cluster/adapter/webhook/postgres_webhook_integration_test.go:588]: done

File: cmd/main.go (hunk 19 above)
Comment (limak9182) [reply cmd/main.go]: removed

File: pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go (hunk 20 above)
Comment (limak9182) [reply pkg/postgresql/cluster/adapter/webhook/postgrescluster_validation.go:46]: removed

File: pkg/splunk/enterprise/validation/context.go (hunk 21 above)
Comment (limak9182) [reply pkg/splunk/enterprise/validation/context.go]: removed
